main/internal/implementations/dragon_drop/http_dragondrop_managed_visualization.go

```diff
@@ -1,3 +1,408 @@
+func getTerraformFootprintDataByFile(filename string) (TerraformFootprintData, error) {
+	mainFileContent, err := readFile(filename, "current_cloud")
+	if err != nil {
+		return TerraformFootprintData{}, err
+	}
+
+	inputHCLFile, hclDiag := hclwrite.ParseConfig(
+		mainFileContent,
+		"placeholder.tf",
+		hcl.Pos{Line: 0, Column: 0, Byte: 0},
+	)
+	if hclDiag.HasErrors() {
+		return TerraformFootprintData{}, fmt.Errorf("error parsing HCL file: %s", hclDiag.Error())
+	}
+
+	terraform := inputHCLFile.Body().FirstMatchingBlock("terraform", nil)
+	terraformVersionAttribute := terraform.Body().GetAttribute("required_version")
+	terraformVersionValue := string(terraformVersionAttribute.BuildTokens(nil).Bytes())
+
+	re := regexp.MustCompile(`"(.*)"`)
+	matches := re.FindStringSubmatch(terraformVersionValue)
+
+	if len(matches) < 2 {
+		return TerraformFootprintData{}, fmt.Errorf("[error parsing terraform version value]%w", err)
+	}
+
+	terraformVersion := fmt.Sprintf(`{"%s": 1}`, matches[1])
+
+	requiredProviders := terraform.Body().FirstMatchingBlock("required_providers", nil)
+	providerAttribute, err := getProviderAttribute(requiredProviders)
+	if err != nil {
+		return TerraformFootprintData{}, nil
+	}
+
+	providerVersionValue := getVersionFromProviderAttribute(providerAttribute)
+	providerVersion := fmt.Sprintf(`{"%s": 1}`, providerVersionValue)
+
+	return TerraformFootprintData{
+		VersionsTFProviders: providerVersion,
+		VersionsTF:          terraformVersion,
+	}, nil
+}
+
+// getProviderAttribute gets the provider attribute from the required_providers block searching for aws, azurerm or google
+func getProviderAttribute(requiredProviders *hclwrite.Block) (*hclwrite.Attribute, error) {
+	awsProvider := requiredProviders.Body().GetAttribute("aws")
+	if awsProvider != nil {
+		return awsProvider, nil
+	}
+
+	azurermProvider := requiredProviders.Body().GetAttribute("azurerm")
+	if azurermProvider != nil {
+		return azurermProvider, nil
+	}
+
+	googleProvider := requiredProviders.Body().GetAttribute("google")
+	if googleProvider != nil {
+		return googleProvider, nil
+	}
+
+	return nil, errors.New("[provider not supported]")
+}
+
+// getVersionFromProviderAttribute gets the version from the provider attribute passed as a parameter
+func getVersionFromProviderAttribute(attribute *hclwrite.Attribute) string {
+	tokens := attribute.BuildTokens(nil)
+	versionIndex := -1
+	for i, token := range tokens {
+		if string(token.Bytes) == "version" {
+			versionIndex = i
+			break
+		}
+	}
+
+	versionValueBytes := make([]byte, 0)
+	beginToGetToken := false
+	for _, token := range tokens[versionIndex:] {
+		if beginToGetToken {
+			versionValueBytes = append(versionValueBytes, token.Bytes...)
+			break
+		}
+		if string(token.Bytes) == "\"" {
+			beginToGetToken = true
+		}
+	}
+
+	versionValue := string(versionValueBytes)
+	versionValue = strings.Trim(versionValue, "~>")
```

Comment:
Some other edge cases for the provider version which need to be supported:
- "=", ">="

Reply:
May be worth unit testing this function separately with the new complexity